Mathlib/Topology/LocallyClosed.lean

```diff
@@ -0,0 +1,216 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Topology.LocalAtTarget
+
+/-!
+# Locally closed sets
+
+## Main results
+- `IsLocallyClosed`: Predicate saying that a set is locally closed
+- `isLocallyClosed_tfae`:
+  A set `s` is locally closed if one of the equivalent conditions below hold
+  1. It is the intersection of some open set and some closed set.
+  2. The coborder `(closure s \ s)ᶜ` is open.
+  3. `s` is closed in some neighborhood of `x` for all `x ∈ s`.
+  4. Every `x ∈ s` has some open neighborhood `U` such that `U ∩ closure s ⊆ s`.
+  5. `s` is open in the closure of `s`.
+
+-/
+
+variable {α β : Type*} [TopologicalSpace α] [TopologicalSpace β] {s t : Set α} {f : α → β}
+
+open scoped Topology Set.Notation
+
+/--
+The coborder is defined as the complement of `closure s \ s`,
+or the union of `s` and the complement of `∂(s)`.
+This is the largest set such that `s` is closed in, and `s` is locally closed if and only if
+`coborder s` is open.
+
+This is unnamed in the literature, and this name is due to the fact that `coborder s = (border sᶜ)ᶜ`
+where `border s = s \ interior s` is the border in the sense of Hausdorff.
+-/
+def coborder (s : Set α) : Set α :=
+  (closure s \ s)ᶜ
+
+lemma subset_coborder :
+    s ⊆ coborder s := by
+  rw [coborder, Set.subset_compl_iff_disjoint_right]
+  exact disjoint_sdiff_self_right
+
+lemma coborder_inter_closure :
+    coborder s ∩ closure s = s := by
+  rw [coborder, ← Set.diff_eq_compl_inter, Set.diff_diff_right_self, Set.inter_eq_right]
+  exact subset_closure
+
+lemma closure_inter_coborder :
+    closure s ∩ coborder s = s := by
+  rw [Set.inter_comm, coborder_inter_closure]
+
+lemma coborder_eq_union_frontier_compl :
+    coborder s = s ∪ (frontier s)ᶜ := by
+  rw [coborder, compl_eq_comm, Set.compl_union, compl_compl, ← Set.diff_eq_compl_inter,
+    ← Set.union_diff_right, Set.union_comm, ← closure_eq_self_union_frontier]
+
+lemma coborder_eq_univ_iff :
+    coborder s = Set.univ ↔ IsClosed s := by
+  simp [coborder, Set.diff_eq_empty, closure_subset_iff_isClosed]
+
+alias ⟨_, IsClosed.coborder_eq⟩ := coborder_eq_univ_iff
+
+lemma coborder_eq_compl_frontier_iff :
+    coborder s = (frontier s)ᶜ ↔ IsOpen s := by
```

Comment:
Can you add a lemma stating `Disjoint (frontier s) s ↔ IsOpen s`? Then this proof can become
```lean
  simp_rw [coborder_eq_union_frontier_compl, union_eq_right, subset_compl_iff_disjoint_left, 
    disjoint_frontier_iff_isOpen]
```